workflows/illumina.nf

```diff
@@ -107,16 +166,102 @@ workflow ILLUMINA {
   ch_versions = ch_versions.mix(POST_TABLE2ASN.out.versions)
 
-  BLAST_BLASTN(IRMA.out.consensus, BLAST_MAKEBLASTDB.out.db)
-  ch_versions = ch_versions.mix(BLAST_BLASTN.out.versions.first().ifEmpty(null))
+  BLAST_BLASTN_IRMA(IRMA.out.consensus, BLAST_MAKEBLASTDB_NCBI.out.db)
+  ch_versions = ch_versions.mix(BLAST_BLASTN_IRMA.out.versions.first().ifEmpty(null))
 
-  ch_blast = BLAST_BLASTN.out.txt.collect({ it[1] })
-  SUBTYPING_REPORT(
+  ch_blast = BLAST_BLASTN_IRMA.out.txt.collect({ it[1] })
+  SUBTYPING_REPORT_IRMA_CONSENSUS(
     ZSTD_DECOMPRESS_CSV.out.file,
     ch_blast,
     CHECK_SAMPLE_SHEET.out
   )
-  ch_versions = ch_versions.mix(SUBTYPING_REPORT.out.versions)
+  ch_versions = ch_versions.mix(SUBTYPING_REPORT_IRMA_CONSENSUS.out.versions)
+
+  // Prepare top ncbi accession id for each segment of each sample (id which has top bitscore)
+  PULL_TOP_REF_ID(BLAST_BLASTN_IRMA.out.txt, ZSTD_DECOMPRESS_CSV.out.file)
+  ch_versions = ch_versions.mix(PULL_TOP_REF_ID.out.versions)
+
+  PULL_TOP_REF_ID.out.accession_id
+    .map { it[1] }
+    .splitCsv(header: false, sep:",")
+    .map{ [it[0], it[1], it[2]] }
+    .combine(CAT_ILLUMINA_FASTQ.out.reads.map { [it[0].id, it[1]] }, by: 0)
+    .set { ch_sample_segment } 
+
+  // Pull segment reference sequence for each sample
+  SEQTK_SEQ(ch_sample_segment, ZSTD_DECOMPRESS_FASTA.out.file)
+  ch_versions = ch_versions.mix(SEQTK_SEQ.out.versions)
+
+  // Map reads against segment reference sequences using Minimap2
+  MINIMAP2(SEQTK_SEQ.out.sample_info)
+  ch_versions = ch_versions.mix(MINIMAP2.out.versions)
+
+  // Generate coverage and stats
+  MOSDEPTH_GENOME(MINIMAP2.out.alignment)
+  ch_versions = ch_versions.mix(MOSDEPTH_GENOME.out.versions)
+
+  FREEBAYES(MINIMAP2.out.alignment)
+  ch_versions = ch_versions.mix(FREEBAYES.out.versions)
+
+  BCF_FILTER_FREEBAYES(FREEBAYES.out.vcf, params.major_allele_fraction)
+  ch_versions = ch_versions.mix(BCF_FILTER_FREEBAYES.out.versions)
+  ch_vcf_filter = BCF_FILTER_FREEBAYES.out.vcf
+
+  VCF_FILTER_FRAMESHIFT(ch_vcf_filter)
+  ch_versions = ch_versions.mix(VCF_FILTER_FRAMESHIFT.out.versions)
+
+  BCFTOOLS_STATS(VCF_FILTER_FRAMESHIFT.out.vcf)
+  ch_versions = ch_versions.mix(BCFTOOLS_STATS.out.versions)
+
+  VCF_FILTER_FRAMESHIFT.out.vcf
+    .combine(MOSDEPTH_GENOME.out.bedgz, by: [0, 1, 2]) // combine channels based on sample_name, segment and accession_id
+    .set { ch_bcf_consensus } // ch_bcf_consensus: [sample_name, segment, id, fasta, filt_vcf, mosdepth_per_base]
+
+  COVERAGE_PLOT(ch_bcf_consensus, params.low_coverage)
+  ch_versions = ch_versions.mix(COVERAGE_PLOT.out.versions)
+
+  // Generate consensus sequences
+  BCF_CONSENSUS(ch_bcf_consensus, params.low_coverage)
+  ch_versions = ch_versions.mix(BCF_CONSENSUS.out.versions)
+
+  BCF_CONSENSUS.out.fasta
+    .groupTuple(by: 0)
+    .set { ch_final_consensus }
+
+  CAT_CONSENSUS(ch_final_consensus)
+  ch_versions = ch_versions.mix(CAT_CONSENSUS.out.versions)
+
+  CAT_CONSENSUS.out.fasta
```

Comment:
Would you be able to add VADR annotation after the Bcftools consensus sequences? Annotation of both IRMA and Bcftools consensus sequences might be useful in case there are interesting differences.

Reply:
@peterk87 added VADR annotation after Bcftools consensus. Please approve for testing. Thanks!